src/queries/role.ts

```diff
@@ -11,20 +13,37 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+type Role = AccessibilityRole | TextMatch;
+
+type ByRoleOptions = {
+  name?: TextMatch;
+};
+
+const filterByAccessibleName = (
```

Comment:
Since the method is not an array filter method but rather a predicate let's tweak the name a bit to reduce confusion:

```suggestion
const matchAccessibleName = (
```